politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +392,256 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
 	return &sr, nil
 }
 
+// comments requests all comments on a record from the comments plugin.
+func (p *piPlugin) comments(token []byte) (*comments.GetAllReply, error) {
+	reply, err := p.backend.PluginRead(token, comments.PluginID,
+		comments.CmdGetAll, "")
+	if err != nil {
+		return nil, err
+	}
+	var gar comments.GetAllReply
+	err = json.Unmarshal([]byte(reply), &gar)
+	if err != nil {
+		return nil, err
+	}
+	return &gar, nil
+}
+
+// isInCommentTree returns whether the leafID is part of the provided comment
+// tree. A leaf is considered to be part of the tree if the leaf is a child of
+// the root or the leaf references the root itself.
+func isInCommentTree(rootID, leafID uint32, cs []comments.Comment) bool {
+	if leafID == rootID {
+		return true
+	}
+	// Convert comments slice to a map
+	commentsMap := make(map[uint32]comments.Comment, len(cs))
+	for _, c := range cs {
+		commentsMap[c.CommentID] = c
+	}
+
+	// Travel the comment tree searching for rootID.
+	// If we reach the tree haed without visiting the provided
+	// rootID then it is not the root.
+	current := commentsMap[leafID]
+	for current.ParentID != 0 {
```

Comment:
Your documentation could be a little more clear here.  Assume a developer is reading this who is only minimally familiar with the politeia repo.  Two things seem unclear to me:

1. You say you're traveling the comment tree, but you don't say how you're traveling the tree.  There are all kinds of algorithms for traversing a tree.
2. You talk about reaching the tree head, but it's not clear what indicates that a node is the tree head.  I would have to read through the comments API to make the connection that a `ParentID` of 0 means that you've reached the tree head.


Example of how you this documentation could be more clear:
Start with the provided comment leaf and traverse the comment tree up until either the provided root ID is found or we reach the tree head.  The tree head will have a comment ID of 0.  

You should aim to make your code dead simple to read through for a developer with minimal familiarity.

Reply:
💯